sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -353,7 +392,32 @@ static struct snd_soc_dai_link *sof_card_dai_links_create(struct device *dev,
 		links[id].no_pcm = 1;
 		id++;
 	}
+	if (sof_rt5682_quirk & SOF_SPEAKER_AMP_PRESENT) {
 
+		links[id].name = devm_kasprintf(dev, GFP_KERNEL,
+					"SSP%d-Codec", ssp_spk);
+		links[id].id = id;
+		links[id].codec_name = MAXIM_SPEAKER_DEV_NAME;
+		links[id].codec_dai_name = MAXIM_SPEAKER_AMP_DAI;
+		links[id].platforms = platform_component;
+		links[id].num_platforms = ARRAY_SIZE(platform_component);
+		links[id].init = speaker_codec_init,
+		links[id].dai_fmt = SND_SOC_DAIFMT_I2S |
+				SND_SOC_DAIFMT_NB_NF |
+				SND_SOC_DAIFMT_CBS_CFS;
```

Comment:
Do we need to set dai_fmt on machine driver? I think it is set by topology.

Reply:
yes, no need for dai_fmt here.